.github/workflows/ci.yml

```diff
@@ -274,9 +274,22 @@ jobs:
         # work properly.
         METADATA_DOCS_RS_RUSTDOC_ARGS="$(cargo metadata --format-version 1 | \
           jq -r ".packages[] | select(.name == \"zerocopy\").metadata.docs.rs.\"rustdoc-args\".[]" | tr '\n' ' ')"
+        # We invoke `cargo deadlinks` instead of `cargo doc`; `cargo deadlinks`
+        # is a link-checking wrapper around `cargo doc`.
         export RUSTDOCFLAGS="${{ matrix.toolchain == 'nightly' && '-Z unstable-options --document-hidden-items' || '' }} $RUSTDOCFLAGS $METADATA_DOCS_RS_RUSTDOC_ARGS"
         ./cargo.sh +${{ matrix.toolchain }} doc --document-private-items --package ${{ matrix.crate }} ${{ matrix.features }}
 
+    - name: Check dead links
+      run: |
+        cargo install -q cargo-deadlinks --version 0.8.1
+        METADATA_DOCS_RS_RUSTDOC_ARGS="$(cargo metadata --format-version 1 | \
+          jq -r ".packages[] | select(.name == \"zerocopy\").metadata.docs.rs.\"rustdoc-args\".[]" | tr '\n' ' ')"
+        # We invoke `cargo deadlinks` instead of `cargo doc`; `cargo deadlinks`
+        # is a link-checking wrapper around `cargo doc`.
+        export RUSTDOCFLAGS="${{ matrix.toolchain == 'nightly' && '-Z unstable-options --document-hidden-items' || '' }} $RUSTDOCFLAGS $METADATA_DOCS_RS_RUSTDOC_ARGS"
+        ./cargo.sh +${{ matrix.toolchain }} deadlinks --check-intra-doc-links -- --document-private-items --package ${{ matrix.crate }} ${{ matrix.features }}
+      if: matrix.toolchain != 'msrv'
```

Comment:
Explain why?

Also, I'd recommend just nightly. Runs fewer jobs and the behavior of this is very unlikely to differ by toolchain. Notably, our docs.rs docs are only built from nightly.

Reply:
The MSRV of this version of deadlinks is higher than our MSRV.

Restricting to nightly seems reasonable, though.